homeassistant/helpers/event.py

```diff
@@ -380,23 +377,27 @@ def _remove_empty_listener() -> None:
     """Remove a listener that does nothing."""
 
 
-@callback  # type: ignore[arg-type]  # mypy bug?
+@callback
 def _remove_listener(
     hass: HomeAssistant,
-    listeners_key: HassKey[Callable[[], None]],
+    tracker: _KeyedEventTracker[_TypedDictT],
     keys: Iterable[str],
     job: HassJob[[Event[_TypedDictT]], Any],
     callbacks: dict[str, list[HassJob[[Event[_TypedDictT]], Any]]],
 ) -> None:
     """Remove listener."""
-    for key in keys:
-        callbacks[key].remove(job)
-        if len(callbacks[key]) == 0:
-            del callbacks[key]
+    if isinstance(keys, str):
```

Comment:
is this a new feature? A `str` is not an `Iterable

Reply:
I split that so it wasn't wrapped in a list anymore since 99% of the time it's actually a str.  I need to fix the type

Reply:
https://github.com/home-assistant/core/pull/118010/files#diff-4dcfd2a5631cf11264898529f88c364230a82a3a1dc9406f802f267ae3723437L444

Reply:
I should change that to a tuple wrapping instead than we don't have the branch here

Reply:
cleaned that up a bit